lib/board.rb

```diff
@@ -0,0 +1,65 @@
+class Board
+  attr_reader :grid
+
+  SPACES = 9
+  ROWS = [[0, 1, 2], [3, 4, 5], [6, 7, 8]].freeze
+  COLUMNS = [[0, 3, 6], [1, 4, 7], [2, 5, 8]].freeze
+  DIAGONALS = [[0, 4, 8], [2, 4, 6]].freeze
+
+  def initialize
+    @grid = create
+  end
+
+  def print
+    puts "|#{@grid[0]}|" + "#{@grid[1]}|" + "#{@grid[2]}|"
+    puts "|#{@grid[3]}|" + "#{@grid[4]}|" + "#{@grid[5]}|"
+    puts "|#{@grid[6]}|" + "#{@grid[7]}|" + "#{@grid[8]}|"
+  end
+
+  def space_exist?(space)
+    space >= 0 && space < 9
+  end
+
+  def space_is_free?(space)
+    space_exist?(space) && @grid[space] == '-'
+  end
+
+  def winning_move?
+    win?(ROWS) || win?(COLUMNS) || win?(DIAGONALS)
+  end
+
+  def drawing_move?
+    !winning_move? && !@grid.include?('-')
+  end
+
+  private
+
+  def create(n = SPACES, val = '-')
+    Array.new(n, val)
+  end
+
+  def win?(type)
```

Comment:
Abstraction: Is it possible to make this code not need to have a special case for checking diagonal wins? This would make it easier to read and easier to change.